src/Entity/BannerSelection/Campaign.php

```diff
@@ -12,17 +12,19 @@ class Campaign {
 	private $identifier;
 	private $start;
 	private $end;
+	private $impressionThreshold;
 
 	/**
 	 * @var Bucket[]
 	 */
 	private $buckets;
 
-	public function __construct( string $identifier, \DateTime $start, \DateTime $end, array $buckets ) {
+	public function __construct( string $identifier, \DateTime $start, \DateTime $end, array $buckets, ImpressionThresholdInterface $impressionThreshold ) {
```

Comment:
Buckets should be the last param to allow for variadic arguments instead of untyped array.

Could you update the commit message with a reason why you decided to bring it back into Campaign?

Reply:
It's outdated now cause I just refactored this whole bit. If you have the time, feel free to look at the latest commit to this branch. I will add the spread operator for buckets in another commit.